docs/markdown/Getting Started/getting-started/installation.md

```diff
@@ -6,40 +6,63 @@ createdAt: "2020-02-21T17:44:53.022Z"
+
+If you have difficulty installing Pants, see our [getting help](doc:getting-help) for community resources to help you resolve your issue.
+
+> 👍 Upgrading Pants
+> 
+> The `pants` launcher binary will automatically install and use the Pants version specified in `pants.toml`, so upgrading Pants in a repo is as simple as editing `pants_version` in that file.
+>
+> To upgrade the `pants` launcher binary itself, run
+> ```
+> SCIE_BOOT=update pants
+> ```
 
-To streamline this, we provide a script that will create a minimal `pants.toml` file specifying the latest released version of Pants, download, and run the `./pants` launch script.
+Running Pants from unreleased builds
+------------------------------------
+
+To use an unreleased build of Pants from the [pantsbuild/pants](https://github.com/pantsbuild/pants) main branch, locate the main branch SHA, set `PANTS_SHA=<SHA>` in the environment, and run `pants` as usual:
 
 ```
-/bin/bash -c "$(curl -fsSL https://static.pantsbuild.org/setup/one_step_setup.sh)" 
+PANTS_SHA=8553e8cbc5a1d9da3f84dcfc5e7bf3139847fb5f pants --version
 ```
 
-If the installation process was successful, you will see `Pants was installed successfully!` echoed to your terminal. 
+Running Pants from sources
+--------------------------
 
-If you had difficulty installing Pants, see our [getting help](doc:getting-help) for community resources to help you resolve your issue.
+See [here](doc:running-pants-from-sources) for instructions on how to run Pants directly from its [sources](https://github.com/pantsbuild/pants).
 
+This is useful when making changes directly to Pants, to see how those changes impact your repo.
 
-> 📘 Add `./pants` to version control
-> 
-> You should check the `./pants` script into your repo so that all users can easily run Pants.
 
-> 👍 Upgrading Pants
+> 🚧 The old `./pants` script
+>
+> Before the creation of the `pants` launcher binary, the recommended way of installing Pants was to check a `./pants` launcher script into each repo. This script required an external Python interpreter, and was prone to errors and issues related to discovery and use of this interpreter. 
+> 
+> The `pants` launcher binary uses an embedded interpreter and does not rely on one being present on the system (although if your repo contains Python code then it naturally requires a Python interpreter).
 > 
-> The `./pants` script will automatically install and use the Pants version specified in `pants.toml`, so upgrading Pants is as simple as editing `pants_version` in that file.
+> We strongly recommend removing the `./pants` script from your repo and using the `pants` binary instead. You can keep a simple `./pants` script that delegates to `pants` to ease the transition. However, if you do need to continue to use the old installation method for some reason, it is described [here](doc:manual-installation). But please [let us know](doc:getting-help) so we can accommodate your use case in the launcher binary.
```

Comment:
What we're trying to get away from is having that script do all the laborious bootstrapping work it currently does. But you're certainly welcome to have a `./pants` script that delegates to `pants` and, if it's missing, either tells users how to install that, or actually installs it. 

We haven't (yet?) created one because:
A) we're not sure we want to recommend that for new users
B) we're not sure what such a script should do. E.g., should it install for you, or just tell you how to install? How would it manage adding `~/bin` to your PATH?

Also, and I have not yet added this to this PR, but macOS users can also install via `brew`, so that's a whole other installation mechanism. 

So we're not sure there's a one-size-fits-all `./pants` script here, and for now it may make sense for repos to write what makes sense for them.

Reply:
I second leaving this to end users to come up with, as it'll be a small-ish script if used at all, adapted to each situation. We could provide an example or two in docs for what it could look like for a few common cases, perhaps.